[HOPSWORKS-2166] Logging with filebeat daemonset and airgapped installation support

javierdlrm opened the pull request.

SirOibaf commented on Karamelfile:
```diff
@@ -28,6 +28,13 @@ dependencies:
       - kube-hops::addons      
       - hops::docker_registry
 
+  - recipe: kube-hops::filebeat
+    global:
+      - kube-hops::node
+      - kube-hops::master
+      - kube-hops::addons      
+      - hops::docker_registry
```
As you include the recipe in another recipe (correctly), Karamel is not aware of the existence of the filebeat recipe. Hence you don't need this part.

javierdlrm replied:
fixed

SirOibaf commented on templates/default/filebeat-serving.yml.erb:
```diff
@@ -0,0 +1,123 @@
+        - type: kubernetes
+          node: ${NODE_NAME}
+          templates:
+            - condition.has_fields: ['kubernetes.labels.serving_hops_works/id']
```
Isn't this picking up the logs for every container running? Including the control plane, Jupyter and KfServing control pods?

javierdlrm replied:
I added the `serving.hops.works/*` labels for model serving. It will pick up the logs for every container inside a pod with a `serving.hops.works/id` label, which are kfserving inference service pods and the current serving deployments we have. We don't add these labels to Jupyter pods.

SirOibaf commented on templates/default/filebeat-serving.yml.erb:
```diff
@@ -0,0 +1,123 @@
+      dnsPolicy: ClusterFirstWithHostNet
+      containers:
+        - name: filebeat
+          image: docker.elastic.co/beats/filebeat:<%= node['kube-hops']['filebeat']['version'] %>
```
No you can't do that. 1st because of licensing and 2nd because we need airgapped deployments to work. 
You need to build a Docker image including filebeat from opendistro and then publish it as a tar file and upload it to the registry.

I think there was a comment with the same recommendation also in the previous PR.

javierdlrm replied:
I'll upgrade filebeat in docker-images repo to 7.9.3 and use that one.

tkakantousis commented:
@javierdlrm @SirOibaf An update on addressing the comments and approving the PR?

javierdlrm commented:
> @javierdlrm @SirOibaf Any update on addressing the comments and approving the PR?

I'll upgrade filebeat in docker-images repo to 7.9.3 and add support for airgapped installations by loading the images from a `kfserving.tar`  in `repo.hops.works`
